server/ws/websockets.go

```diff
@@ -99,9 +103,11 @@ func (ws *Server) handleWebSocketOnChange(w http.ResponseWriter, r *http.Request
 		client.Close()
 	}()
 
+	userID := r.Header.Get("Mattermost-User-Id")
```

Comment:
Also we need to check if the authentication method is mattermost, if not we can end up with a security hole here.